createLink TypeScript error when passing Link props to underlying component with composite enabled

### Which project does this relate to?

Router

### Describe the bug

When using createLink 5.6.2 with @mantine/core's Button component and setting activeProps on the Button, I receive the following TypeScript error:

client/Components/References/References.tsx:84:14 - error TS4023: Exported variable 'ButtonLink' has or is using name 'MakeOptionalSearchParams' from external module "/home/jan/dev/plato/node_modules/.pnpm/@tanstack+react-router@1.58.15_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/@tanstack/react-router/dist/esm/link" but cannot be named.

84 export const ButtonLink = createLink(


```typescript
import '@mantine/core/styles.css';
import { Button, ButtonProps, MantineProvider } from '@mantine/core';

export const ButtonLink = createLink(
  React.forwardRef(
    (
      props: CreateLinkProps & ButtonProps & { title?: string },
      ref: React.Ref<HTMLAnchorElement>
    ) => {
      return (
        <Button
          {...props}
          ref={ref}
          component={Link}
          activeProps={{ ...props.activeProps, 'data-active': true }}
        />
      );
    }
  )
);
```

I found out that this is correlated to exporting the created Link and using the composite setting in tsconfig. I disabled the setting for now, but I wouldn't be surprised if someone might need it in the future.


### Your Example Website or App

https://stackblitz.com/edit/github-ywacnp?file=src%2Fmain.tsx

### Steps to Reproduce the Bug or Issue

1. Go to the StackBlitz example
2. See error (file might have to be resaved/ remove export -> save -> add export -> save to see the error; error is persistend in my local VSCode)
3. Remove composite setting in tsconfig
4. Error gone and won't come back after doing step 2 again

### Expected behavior

No type error even with composite enabled

### Screenshots or Videos

_No response_

### Platform

- OS: WSL Ubuntu 24.04
- Browser: Brave
- Version: 1.17.119
- Typescript: 5.6.2


### Additional context

_No response_

--- a/packages/react-router/src/index.tsx
+++ b/packages/react-router/src/index.tsx
@@ -74,6 +74,7 @@ export type {
   LinkProps,
   LinkComponent,
   CreateLinkProps,
+  MakeOptionalPathParams,
 } from './link'
 
 export type { ParsedLocation } from './location'

--- a/packages/react-router/src/link.tsx
+++ b/packages/react-router/src/link.tsx
@@ -356,7 +356,7 @@ type ResolveRelativeToParams<
         TToParams
       >
 
-interface MakeOptionalSearchParams<
+export interface MakeOptionalSearchParams<
   in out TRouter extends AnyRouter,
   in out TFrom,
   in out TTo,
@@ -364,7 +364,7 @@ interface MakeOptionalSearchParams<
   search?: true | (ParamsReducer<TRouter, 'SEARCH', TFrom, TTo> & {})
 }
 
-interface MakeOptionalPathParams<
+export interface MakeOptionalPathParams<
   in out TRouter extends AnyRouter,
   in out TFrom,
   in out TTo,

